server/plugin/api.go

```diff
@@ -749,6 +750,35 @@ func (p *Plugin) handleGetGitRepositories(w http.ResponseWriter, r *http.Request
 	p.writeJSON(w, response.Value)
 }
 
+func (p *Plugin) handleGetGitRepositoryBranches(w http.ResponseWriter, r *http.Request) {
+	mattermostUserID := r.Header.Get(constants.HeaderMattermostUserID)
+
+	pathParams := mux.Vars(r)
+	organization := pathParams[constants.PathParamOrganization]
+	project := pathParams[constants.PathParamProject]
+	repository := pathParams[constants.PathParamRepository]
+
+	if len(strings.TrimSpace(organization)) == 0 || len(strings.TrimSpace(project)) == 0 || len(strings.TrimSpace(repository)) == 0 {
+		p.API.LogError(constants.ErrorRepositoryPathParam)
+		p.handleError(w, r, &serializers.Error{Code: http.StatusBadRequest, Message: constants.ErrorRepositoryPathParam})
+		return
+	}
+
+	response, statusCode, err := p.Client.GetGitRepositoryBranches(organization, project, repository, mattermostUserID)
+	if err != nil {
+		p.API.LogError("Error in fetching git repository branches", err.Error())
+		p.handleError(w, r, &serializers.Error{Code: statusCode, Message: err.Error()})
+		return
+	}
+
+	// Azure DevOps return branch name as "refs/heads/<branch-name>", but we need to use only "<branch-name>" so, remove unused part from the name
+	for index, value := range response.Value {
+		response.Value[index].Name = value.Name[11:]
+	}
```

Comment:
```suggestion
	for _, value := range response.Value {
		value.Name = value.Name[11:]
	}
```

Reply:
check if this works
also, are we always sure that this length is greater than 11?

Reply:
yes it will be always >= 11, still added the checks